average.js

```diff
@@ -1,6 +1,12 @@
-
 function average(numbers) {
-  return numbers.reduce((p, c)=> p + c, 0) / numbers.length;
+  var n =numbers.length
+  return numbers.reduce((p,c)=>{
+    if (isNaN(c))
+    {
+      n-=1
```

Comment:
You are doing the summing and elimination of NaN in a single pass. This is good in many situations.
However, consider the trade-off as well: Making the summing and counting NaNs in separate loops enables them to run in parallel, resulting in faster turnaround-time for large data-sets

Reply:
I have implemented the asked changes.

Reply:
If I am using another loop to count then variable complexity is exceeding.

Reply:
I got the complexity down upto 4 but max allowed is 3.